simulation/traffic_simulator/include/traffic_simulator/data_type/routing_configuration.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright 2015 TIER IV, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TRAFFIC_SIMULATOR__DATA_TYPE__ROUTING_CONFIGURATIONS_HPP_
+#define TRAFFIC_SIMULATOR__DATA_TYPE__ROUTING_CONFIGURATIONS_HPP_
+#include <iostream>
+#include <traffic_simulator/data_type/routing_graph_type.hpp>
+
+namespace traffic_simulator
+{
+struct RoutingConfiguration
+{
+  bool allow_lane_change = false;
+  traffic_simulator::RoutingGraphType routing_graph_type =
+    traffic_simulator::RoutingGraphType::VEHICLE;
+
+  bool operator==(const RoutingConfiguration & routing_configuration) const
+  {
+    return allow_lane_change == routing_configuration.allow_lane_change &&
+           routing_graph_type == routing_configuration.routing_graph_type;
+  }
+
+  friend std::ostream & operator<<(std::ostream & os, const RoutingConfiguration & rc)
+  {
+    os << "{ allow_lane_change: " << (rc.allow_lane_change ? "true" : "false")
+       << ", routing_graph_type: " << to_string(rc.routing_graph_type) << " }";
+    return os;
+  }
+};
+}  // namespace traffic_simulator
+
+namespace std
+{
+template <>
+struct hash<traffic_simulator::RoutingConfiguration>
+{
+  std::size_t operator()(const traffic_simulator::RoutingConfiguration & config) const
+  {
+    auto h1 = std::hash<bool>{}(config.allow_lane_change);
+    auto h2 = std::hash<std::uint8_t>{}(static_cast<std::uint8_t>(config.routing_graph_type));
+    return h1 ^ (h2 << 1);
+  }
+};
+}  // namespace std
```

Comment:
Is it necessary?
If you plan to use them as keys for unordered associative containers (unordered_map / unordered_multimap / unordered_set / unordered_multiset), then withdraw this suggestion.

Reply:
Oh, it's my mistake, thank you!
I deleted it in [`e91b235` (#1458)](https://github.com/tier4/scenario_simulator_v2/pull/1458/commits/e91b2356ce77ae1a9013aa9e4946831c428ba485)